lib/rodauth/features/jwt.rb

```diff
@@ -37,12 +51,8 @@ def session
       return @session if defined?(@session)
       return super unless use_jwt?
 
-      @session = if jwt_token
-        s = {}
-        jwt_payload.each do |k,v|
-          s[k.to_sym] = v
-        end
-        s
+      @session = if jwt_token && data = jwt_payload['data']
+        JSON.parse(JSON.fast_generate(data), :symbolize_names=>true)
```

Comment:
Replacing a shallow symbolization with a deep one. Unfortunately, ruby-jwt doesn't let you "punch down" the `:symbolize_names` option on decode. I think we can get away with `.fast_generate` here because we know the data was recently acceptable JSON, i.e. no cycles.